common/persistence/client/fault_injection.go

```diff
@@ -162,7 +167,9 @@ func (d *FaultInjectionDataStoreFactory) NewShardStore() (persistence.ShardStore
 		if err != nil {
 			return nil, err
 		}
-		d.ShardStore, err = NewFaultInjectionShardStore(d.ErrorGenerator.Rate(), baseFactory)
+		// targetedShardStoreFaults is a list of faults that are only injected for the shard store. This is optional.
+		targetedShardStoreFaults := d.config.Targets["ShardStore"]
```

Comment:
Is the target config only works for ShardStore? I don't see this for other stores, for example TaskStore. It is not clear from the config itself.